bokehjs/test/unit/core/util/array.ts

```diff
@@ -111,29 +112,27 @@ describe("core/util/array module", () => {
   })
 
   it("linspace should return an array of a given length between two given numbers", () => {
-    expect(array.linspace(2, 3, 5)).to.be.equal([ 2, 2.25, 2.5, 2.75, 3 ])
+    expect(array.linspace(2, 3, 5)).to.be.equal([2, 2.25, 2.5, 2.75, 3])
     expect(array.linspace(2, 3).length).to.be.equal(100)
     expect(array.linspace(2, 3, 0)).to.be.equal([])
-    expect(array.linspace(2, 3, NaN)).to.be.instanceof(RangeError)
-    expect(array.linspace(2, 3, null)).to.be.equal([null])
+    expect(() => array.linspace(2, 3, NaN)).to.throw(RangeError)
     expect(array.linspace(NaN, NaN).length).to.be.equal(100)
   })
 
   it("transpose should return the given array with transposed axes", () => {
     expect(array.transpose([[1, 2], [3, 4]])).to.be.equal([[1, 3], [2, 4]])
     expect(array.transpose([[1, 2, 3, 4], [5, 6, 7, 8]])).to.be.equal([[1, 5], [2, 6], [3, 7], [4, 8]])
-    expect(array.transpose([[1, 2, 3, 4], [5, 6, 7]])).to.be.equal([[1, 5], [2, 6], [3, 7], [4, undefined]])
     expect(array.transpose([[1, null, 3, undefined], [NaN]])).to.be.equal([[1, NaN], [null, undefined], [3, undefined], [undefined, undefined]])
   })
 
   it("remove_at should remove an item from an array at the specified index", () => {
     expect(array.remove_at([1, 2, 3, 4], 1)).to.be.equal([1, 3, 4])
     expect(array.remove_at([1, 2, 3, 4], 5)).to.be.equal([1, 2, 3, 4])
-    expect(array.remove_at([NaN], NaN)).to.be.equal([])
+    expect(() => array.remove_at([NaN], NaN)).to.throw(AssertionError)
   })
 
-  it("clear should return an empty array", () => {
-    expect(array.clear([1, 2, 3, 4])).to.be.equal([])
+  it("clear should return nothing", () => {
+    expect(array.clear([1, 2, 3, 4])).to.be.equal(undefined)
```

Comment:
Actually, this is not what I meant with me previous comment. The test should verify that this function actually does anything useful, something like:
```ts
const arr = [1, 2, 3, 4]
array.clear(arr)
expect(arr).to.be.equal([]) // or
expect(arr.length).to.be.equal(0)
```
Given that we heavily rely to TypeScript, then testing whether void function returns `undefined` is unnecessary.